wallet-new/integration/Functions.hs

```diff
@@ -35,454 +37,446 @@ import qualified Pos.Wallet.Web.ClientTypes.Types as V0
 
-    -- Modify wallet state accordingly.
-    pure $ ws
-        & transactions  <>~ [(accountSource, result)]
-        & actionsNum    +~ 1
+            let transaction :: Transaction
+                transaction = accTransaction ^. _2
 
-  where
-    createNewPayment :: PaymentSource -> [PaymentDistribution] -> Payment
-    createNewPayment ps pd = Payment
-        { pmtSource           = ps
-        , pmtDestinations     = fromList pd
-        , pmtGroupingPolicy   = Nothing
-        -- ^ Simple for now.
-        , pmtSpendingPassword = Nothing
-        }
+            let walletId :: WalletId
+                walletId = accWalletId txsAccount
 
+            let accountIndex :: AccountIndex
+                accountIndex = accIndex txsAccount
 
-runAction wc ws GetTransaction  = do
-    let txs = ws ^. transactions
+            log $ "getTransactionIndex: " <> show walletId <> ", " <> show accountIndex
+            result  <-  respToRes $ getTransactionIndex
+                                        wc
+                                        (Just walletId)
+                                        (Just accountIndex)
+                                        Nothing
+                                        Nothing
+                                        Nothing
 
-    -- We need to have transactions in order to test this endpoint.
-    guard (not (null txs))
+            -- First check we have results
+            checkInvariant
+                (not (null result))
+                (LocalTransactionsDiffer result (map snd txs))
 
-    -- We choose from the existing transactions.
-    accTransaction  <- pickRandomElement txs
+            -- Then check if the transaction exists in the history
+            checkInvariant
+                (transaction `elem` result)
+                (LocalTransactionMissing transaction result)
 
-    let txsAccount :: Account
-        txsAccount = accTransaction ^. _1
-
-    let transaction :: Transaction
-        transaction = accTransaction ^. _2
-
-    let walletId :: WalletId
-        walletId = accWalletId txsAccount
-
-    let accountIndex :: AccountIndex
-        accountIndex = accIndex txsAccount
-
-    result  <-  respToRes $ getTransactionIndex
-                                wc
-                                (Just walletId)
-                                (Just accountIndex)
-                                Nothing
-                                Nothing
-                                Nothing
-
-    -- First check we have results
-    checkInvariant
-        (not (null result))
-        (LocalTransactionsDiffer result)
-
-    -- Then check if the transaction exists in the history
-    checkInvariant
-        (transaction `elem` result)
-        (LocalTransactionMissing transaction result)
-
-    -- Modify wallet state accordingly.
-    pure $ ws
-        & actionsNum +~ 1
+    -- increment successful actions
+    log "Success!"
+    actionsNum += 1
```

Comment:
The `actionsNum += 1` modification was common to all actions, so I factored it out.